### ISSUE TITLE
Unable to click on floating image next to a to-do document list

### ISSUE THREAD

**Issue by niegowski** [ckeditor/ckeditor5#15115]
## 📝 Provide detailed reproduction steps (if any)

![2023-10-02 19 58 48](https://github.com/ckeditor/ckeditor5/assets/1232187/ae017d09-538c-4670-b2cc-77d533a2a3b4)

### ❓ Possible solution

Looks like style `position: relative` added to `<li>` elements is affecting pointer events on floating images. It was added to be able to position the checkbox next to a block widget.

---

If you'd like to see this fixed sooner, add a 👍 reaction to this post.

**Comment by Reinmar** [comment]
Is it a regression? We talked last month about similar issues.

**Comment by niegowski** [comment]
Unfortunately, the CSS is shared with legacy to-do lists and affects the same cases while using legacy to-do lists.

**Comment by niegowski** [comment]
What we tried:
* setting `display: flow-root` on `<li>` elements - the text was not floating around floated objects (it was a rectangular shape)
* setting `display: inline-block` on `<li>` elements - was expanding to the full with when wrapping and moving away from the floated image

What finally worked:
Setting `z-index: 1` on image widgets. Also the `z-index: 2` on selected images. Previously it was there with value 1 to make sure the selected inline image always stays on top of its siblings (https://github.com/ckeditor/ckeditor5/issues/9108).

What we noticed is that the same issue appears on floating tables. Probably we should add similar solution but the problem there is that `float:` style is added inline in the HTML (not via stylesheet).
